From 8ccc287f107db19b1602b214e22d916db9678cc9 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 10:18:37 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom pcie: replaced macro's with corresponding variable

Global conf and pci global variable already assigned with device
specific macro's. So replaced the macro's with corresponding variable.

Change-Id: Ia106df0391a8ae873039a58d2867e06a60aa7a96
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 4d9fdc3..d40ec84 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -20,7 +20,6 @@
 #include <psoc_global_conf_regs.h>
 #include <pcie_dbi_regs.h>
 #include <pcie_phy_regs.h>
-#include <pcie_aux_regs.h>
 #include <arcboot_export.h>
 #include <gaudi2/gaudi2_security.h>
 #include <gaudi2/gaudi2_bootrom.h>
@@ -66,6 +65,7 @@
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N);
 }
 
+/* TODO: move this function to device independent common file  */
 static bool hl_pcie_link_up(void)
 {
 	unsigned int prstn, smlh, rdlh, rst_src;
@@ -75,9 +75,9 @@
 	if ((rst_src == HL_COLD_RST) || (rst_src == HL_PRSTN_RST))
 		return false;
 
-	prstn = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN);
-	smlh = regmap_readl(mmPCIE_AUX_SMLH_LINK_UP);
-	rdlh = regmap_readl(mmPCIE_AUX_RDLH_LINK_UP);
+	prstn = regmap_readl(global_conf.prstn_reg);
+	smlh = regmap_readl(pci.aux_rdlh_link_up_reg);
+	rdlh = regmap_readl(pci.aux_smlh_link_up_reg);
 
 	if (prstn && smlh && rdlh)
 		return true;
